src/lib/reasoners/satml_frontend.ml

```diff
@@ -1303,7 +1303,7 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
 
   let assume env gf _dep =
     (* dep currently not used. No unsat-cores in satML yet *)
-    assert (SAT.decision_level env.satml == 0);
+    (* assert (SAT.decision_level env.satml == 0); *)
```

Comment:
I commented these assertions because one test failed and I forgot to put them back; the test `./alt-ergo -o smtlib2 --prelude tests/cram.t/prelude.ae tests/cram.t/postlude.smt2` still fails because of it

Reply:
Yeah because there are `push`'d level now. We should probably be able to expose the "push level" and check that instead (i.e. assertion that the decision level is equal to the length of the increm_guard). Probably as a `SAT.can_assume env.satml` check or some such.

Reply:
Would it be correct not to assert at the decision level `0`, but at `decision_level env`?

Reply:
>expose the "push level"

Both my solution and yours may be incorrect actually. 

```
(assert f) ; decision level = 0, push level = 0
(check-sat) ; decision level = 1, push level = 1 (because we internally push)
(assert f') ; decision level = 1, push level = 0 (because we internally popped when switching back to assert mode)
(push 42) ; decision level = 1, push level = 42
```

Reply:
Maybe a counter for every query?

Reply:
Yeah I realized that after (because for the first `check-sat` it will be `0` as it used to be). The correct test I believe is `decision_level <= push_level` (to also capture both `(check-sat)(push)(check-sat)` and `(push)(check-sat)(pop)(check-sat)`).

Reply:
Here's another solution: at the end of a query, keep track of the current decision level and set it as the latest known consistency level on the `satml_frontend` side. I find it hard to justify keeping the invariant `decision_level <= push_level` for a consistency check (in other words, if we ever raise this assertion during a pr, I'm not sure how relevant it would be for debugging).

Reply:
> I find it hard to justify keeping the invariant decision_level <= push_level for a consistency check (in other words, if we ever raise this assertion during a pr, I'm not sure how relevant it would be for debugging).

This is not just a consistency check; I am not sure the solver will do the right thing in this case. In particular, I am not sure the assumed assertions cannot be forgotten upon later backtracking — in fact, I am pretty sure they will be forgotten (partially) for the default CDCL-Tableaux solver given my current understanding of `lazy_cnf`.

Reply:
(Note that for decisions level between `0` and `push_level` forgetting assumptions upon backtracking is OK because that is exactly the semantics we want for `push` and `pop`)